async/AsyncAgent.lua

```diff
@@ -91,7 +91,7 @@ function AsyncAgent:takeAction(action)
     self.stateBuffer:push(observation)
   end
 
-  return reward, terminal, self.stateBuffer:readAll()
+  return reward, terminal, self.stateBuffer:readAll(), actionTaken
```

Comment:
Could add the offset `actionOffset` before returning so it doesn't need adding to compare `actionTaken ~= action`.